p2p/chain.py

```diff
@@ -302,6 +313,36 @@ class LightChainSyncer(BaseHeaderChainSyncer):
         return head.block_number
 
 
+# TODO move to a utils module
+class ThroughputTracker:
+    """
+    Tracks throughput using an exponential moving average.
+    https://en.wikipedia.org/wiki/Moving_average#Exponential_moving_average
+    """
+    def __init__(self, default_throughput : Union[int, float], smoothing_factor : float) -> None:
+        self._last_start = None
+        self._throughput = float(default_throughput)
+        if 0 < smoothing_factor < 1:
+            self._alpha = smoothing_factor
+        else:
+            raise ValidationError("Smoothing factor of ThroughputTracker must be between 0 and 1")
+
+    def begin_work(self) -> None:
+        if self._last_start is not None:
+            raise ValidationError("Cannot start the ThroughputTracker again without completing it")
+        self._last_start = time.perf_counter()
+
+    def complete_work(self, work_completed: Union[int, float]) -> None:
+        if self._last_start is None:
+            raise ValidationError("Cannot end the ThroughputTracker without starting it")
+        time_elapsed = time.perf_counter() - self._last_start
+        last_throughput = work_completed / time_elapsed
+        self._throughput = (self._throughput * (1 - self._alpha)) + (last_throughput * self._alpha)
```

Comment:
This function body looks like it's missing a line to reset `self._last_start` to `None`.